clang/lib/Sema/SemaChecking.cpp

```diff
@@ -14585,11 +14585,18 @@ void Sema::CheckAddressOfPackedMember(Expr *rhs) {
                      _2, _3, _4));
 }
 
+static ExprResult UsualUnaryConversionsNoPromoteInt(Sema &S, Expr *E) {
+  // Don't promote integer types
+  if (QualType Ty = E->getType(); S.getASTContext().isPromotableIntegerType(Ty))
+    return S.DefaultFunctionArrayLvalueConversion(E);
+  return S.UsualUnaryConversions(E);
```

Comment:
If I'm following correctly, this specifically skips promoting integers... but it allows promoting floats?  I think I'd prefer to factor out the unary-float conversions into a separate function, and call that here, for clarity.